nixos/modules/services/databases/neo4j.nix

```diff
@@ -36,81 +89,412 @@ let
+              this policy. It is recommended to turn this off again after keys
+              have been generated.</para>
+
+              <para>The public certificate is required to be duplicated to the
+              directory holding trusted certificates as defined by the
+              <option>trustedDir</option> option.</para>
+
+              <para>Keys should in general be generated and distributed offline by a
+              trusted certificate authority and not by utilizing this mode.</para>
+            '';
+          };
+
+          baseDirectory = mkOption {
+            type = types.path;
+            default = "${cfg.directories.certificates}/${name}";
+            description = ''
+              <para>The mandatory base directory for cryptographic objects of this
+              policy.</para>
+
+              <para>It is also possible to override each individual
+              configuration with absolute paths. See the
+              <option>privateKey</option> and <option>publicCertificate</option>
+              policy options.</para>
+            '';
+          };
+
+          ciphers = mkOption {
+            type = types.nullOr (types.listOf types.str);
+            default = null;
+            description = ''
+              Restrict the allowed ciphers of this policy to those defined
+              here. The default ciphers are those of the JVM platform.
+            '';
+          };
+
+          clientAuth = mkOption {
+            type = types.enum [ "NONE" "OPTIONAL" "REQUIRE" ];
+            default = "REQUIRE";
+            description = ''
+              The client authentication stance for this policy.
+            '';
+          };
+
+          privateKey = mkOption {
+            type = types.str;
+            default = "private.key";
+            description = ''
+              The name of private PKCS #8 key file for this policy to be found
+              in the <option>baseDirectory</option>, or the absolute path to
+              the key file. It is mandatory that a key can be found or generated.
+            '';
+          };
+
+          publicCertificate = mkOption {
+            type = types.str;
+            default = "public.crt";
+            description = ''
+              <para>The name of public X.509 certificate (chain) file in PEM format
+              for this policy to be found in the <option>baseDirectory</option>,
+              or the absolute path to the certificate file. It is mandatory
+              that a certificate can be found or generated.</para>
+
+              <para>The public certificate is required to be duplicated to the
+              directory holding trusted certificates as defined by the
+              <option>trustedDir</option> option.</para>
+            '';
+          };
+
+          revokedDir = mkOption {
+            type = types.path;
+            default = "${cfg.directories.certificates}/${name}/revoked";
```

Comment:
Currently, when the user sets `baseDirectory` of some policy, the `revokedDir` still uses the "old" default base directory (`"${cfg.directories.certificates}/${name}"`). To have it use the policies `baseDirectory`, you can add `config` to the submodules arguments (next to `name`), and use `default = "${config.baseDirectory}/revoked"`

Same for `trustedDirs`